lib/SIL/IR/SILFunctionType.cpp

```diff
@@ -1738,17 +1773,34 @@ static bool isPseudogeneric(SILDeclRef c) {
 
 /// Update the result type given the foreign error convention that we will be
 /// using.
-static std::pair<AbstractionPattern, CanType> updateResultTypeForForeignError(
-    ForeignErrorConvention convention, CanGenericSignature genericSig,
-    AbstractionPattern origResultType, CanType substFormalResultType) {
+void updateResultTypeForForeignInfo(
+    const ForeignInfo &foreignInfo, CanGenericSignature genericSig,
+    AbstractionPattern &origResultType, CanType &substFormalResultType) {
+  // If there's no error or async convention, the return type is unchanged.
+  if (!foreignInfo.Async && !foreignInfo.Error) {
+    return;
+  }
+  
+  // A foreign async convention means our lowered return type is Void, since
```

Comment:
In the synchronous-error case I've been commenting on, you could end up with (e.g.) `BOOL` here, as dictated by the foreign error convention.